### DISCUSSION TITLE
Using the MTVAL register to store AHB errors

### DISCUSSION THREAD

**Discussion by andrenemat** [Mi-V-Soft-RISC-V/miv-rv32-documentation#22] [Mi-V RV32 Soft Processors]
SW team is asking how they can handle errors on the AHB Lite bus.
If they try accessing an invalid AHB address for example, they are getting a trap. But they are wondering how they can get more information about the failed access (which address, etc.).
Is this information captured in one of the RISC-V registers?

**Comment by TommyMurphyTM1234** [comment]
Whenever investigating traps, it generally makes sense to check the following registers:

* `mcause`
* `mepc`
* `mtval`

which are documented in the The RISC-V Instruction Set Manual
Volume II: Privileged Architecture:

* https://riscv.org/technical/specifications/

I don't know what _"Using the MTVAL register to store AHB errors"_ means.

**Reply by andrenemat** [reply]
Sorry for the Title, originally the question was asked in a different way.

Thanks for the suggestion of the mcause and mepc registers, we had only reviewed the MTVAL.

**Comment by TommyMurphyTM1234** [comment]
You should probably read this so:

* https://mi-v-ecosystem.github.io/SoftConsole-Documentation/SoftConsole-v2022.2-RISC-V/troubleshooting/riscv_trap.html

**Reply by andrenemat** [reply]
Thanks a lot. This is a very detailed document.
